lib/ansible/modules/cloud/vmware/vmware_tag_manager.py

```diff
@@ -218,15 +236,24 @@ def ensure_state(self):
         removed_tags_for_set = False
         for tag in self.tag_names:
             category_obj, category_name, tag_name = None, None, None
-            if ":" in tag:
-                # User specified category
-                category_name, tag_name = tag.split(":", 1)
-                category_obj = self.search_svc_object_by_name(self.category_service, category_name)
-                if not category_obj:
-                    self.module.fail_json(msg="Unable to find the category %s" % category_name)
-            else:
-                # User specified only tag
-                tag_name = tag
+            if isinstance(tag, dict):
+                tag_name = tag.get('tag')
+                category_name = tag.get('category')
+                if category_name:
+                    # User specified category
+                    category_obj = self.search_svc_object_by_name(self.category_service, category_name)
+                    if not category_obj:
+                        self.module.fail_json(msg="Unable to find the category %s" % category_name)
```

Comment:
What happens if user wants to create new Category instead of attaching tag to existing category?

Reply:
User can create a new category using `vmware_category` and assign it to new/ existing tag using `vmware_tag`. `vmware_tag_manager` only takes care of using existing tags and category.